tests/integration/DataMapper/VisitMapperTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace GroupLife\Core\tests\DataMapper;
+
+use GroupLife\Core\Activity;
+use GroupLife\Core\DataMapper\ActivityMapper;
+use GroupLife\Core\DataMapper\LeaderMapper;
+use GroupLife\Core\DataMapper\ScheduleMapper;
+use GroupLife\Core\DataMapper\SubscriptionMapper;
+use GroupLife\Core\DataMapper\VisitMapper;
+use GroupLife\Core\DataMapper\VisitorMapper;
+use GroupLife\Core\Leader;
+use GroupLife\Core\Schedule;
+use GroupLife\Core\Subscription\Membership;
+use GroupLife\Core\Test\TestCaseWithDb;
+use GroupLife\Core\Visitor;
+
+class VisitMapperTest extends TestCaseWithDb
+{
+
+    /**
+     * @throws \Doctrine\DBAL\Exception
+     * @throws \GroupLife\Core\Exception\SavingToDbIsForbidden
+     */
+    public function testInsert()
+    {
+
+        $schedule = new Schedule(
+            [
+                new Schedule\WeekdayRule('Monday', '10:00'),
+                new Schedule\CancelDayRule('2021-01-11', '10:00'),
+                new Schedule\CancelDayRule('2021-01-25', '10:00')
+            ]
+        );
+        $leader = new Leader('Ivan', 'Teacher');
+        $activity = new Activity('Skiing', $schedule, $leader);
+        $visitor = new Visitor('Petr', 'Student');
+        $subscription = new Membership(new \DateTimeImmutable('2021-01-01'), new \DateInterval('P1M'), $visitor);
+        $activityVisits = $activity->subscribe($visitor, $subscription);
+
+        $scheduleMapper = new ScheduleMapper(self::$db);
+        $leaderMapper = new LeaderMapper(self::$db);
+        $visitorMapper = new VisitorMapper(self::$db);
+        $activityMapper = new ActivityMapper(self::$db);
+        $subscriptionMapper = new SubscriptionMapper(self::$db);
+        $visitMapper = new VisitMapper(self::$db);
+
+        $scheduleMapper->insert($schedule);
+        $leaderMapper->insert($leader);
+        $visitorMapper->insert($visitor);
+        $activityMapper->insert($activity);
+        $subscriptionMapper->insert($subscription);
+        $visitMapper->insert($activityVisits);
```

Comment:
When you found yourself making such a long preparation for a test case, it makes sense to think about a [DB fixture](https://stackoverflow.com/questions/12071344/what-are-fixtures-in-programming).